File: PHP/Creational/Singleton/Show/Singleton.php
<?php
namespace TheRayCode\Singleton;

class Singleton
{
    private static $instance;

    // Private constructor to prevent direct instantiation
    private function __construct() {}

    // Public static method to get the singleton instance
    public static function getInstance()
    {
        if (self::$instance === null) {
            self::$instance = new self();
        }

        return self::$instance;
    }

    // Example method to test the singleton instance
    public function showMessage()
    {
        echo "Hello, I am the Singleton instance!<br/>";
    }
}

File: Java/Creational/Singleton/Show/src/Singleton.java
public class Singleton {
    // Static instance of Singleton, ensures it is eagerly created.
    private static final Singleton instance = new Singleton();

    // Private constructor ensures that no other class can instantiate this class.
    private Singleton() {}

    // Public method to provide access to the instance.
    public static Singleton getInstance() {
        return instance;
    }

    // Example method to show that Singleton is working.
    public void showMessage() {
        System.out.println("Hello, I'm a Singleton!");
    }
}
